test/behaviors/t_for_all.clj

```diff
@@ -172,3 +175,87 @@
   (for-all
     [x gen/int]
     x => integer?))
+
+(gen-let [s (gen/return "s")]
+  s => "s")
+
+(fact "you can put gen-let inside of a fact"
+  (gen-let [s (gen/return "s")]
+    s => "s")
+  )
+(fact "gen-let composes generators"
+  (gen-let [i (gen/elements [1 2 3])
+            s (gen/return (str i))]
+    (Integer/parseInt s) => i))
+
+(fact "gen-let composes multiple generators"
+  (gen-let [i  gen/s-pos-int
+            is (gen/vector gen/int i)
+            e  (gen/elements is)]
+    (count is) => i
+    (some #{e} is) => e))
+
+(fact ":let syntax allows for simple functions over generated elements"
+  (gen-let [i gen/pos-int
+            :let [s  (str i)
+                  s2 (str s s)]]
```

Comment:
I'm curious in general if there is a way to write an incorrect let-binding? If so, what would the failure look like? Might be worth checking out

Reply:
About the same as a badly formed `let` by itself: https://gist.github.com/rafaeldff/76b10f1d25949f3d79b0d10c318b3bdb (should be better in Clojure 10). 

I don't know what kind of impact would that CompilerException make on midje's failure count reporting, but I guess it would be the same as for any other badly formed macro expansion (like a bad `(let)` in the middle of a `fact`.